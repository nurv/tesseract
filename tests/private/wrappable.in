wrappable int fact(int v){
	if (v == 0){
		return 1;
	}else{
		return v * fact(v - 1);
	}
}
void traceBefore(String f,Object[] v){
	System.out.println(f + "("+v[0]+")" + "=>");
}
void traceAfter(String f,Object[] v, int result){
	System.out.println(f + "("+v[0]+")" + "<= " + result);
}
%before fact(int) add traceBefore(java.lang.String,[Ljava.lang.Object;);
%after fact(int) add traceAfter(java.lang.String,[Ljava.lang.Object;,int);
fact(5)
%before fact(int) del traceBefore(java.lang.String,[Ljava.lang.Object;);
%after fact(int) del traceAfter(java.lang.String,[Ljava.lang.Object;,int);
fact(5)
ist.leic.pa.utils.TracingManager.trace("fact(int)");
fact(6)
ist.leic.pa.utils.TracingManager.untrace("fact(int)");
fact(6)