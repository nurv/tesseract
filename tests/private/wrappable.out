> > > > > > fact(int)(5)=>
fact(int)(4)=>
fact(int)(3)=>
fact(int)(2)=>
fact(int)(1)=>
fact(int)(0)=>
fact(int)(0)<= 1
fact(int)(1)<= 1
fact(int)(2)<= 2
fact(int)(3)<= 6
fact(int)(4)<= 24
fact(int)(5)<= 120
120
> > > 120
> > //fact(6) =>
//  fact(5) =>
//    fact(4) =>
//      fact(3) =>
//        fact(2) =>
//          fact(1) =>
//            fact(0) =>
//            fact(0) <= 1
//          fact(1) <= 1
//        fact(2) <= 2
//      fact(3) <= 6
//    fact(4) <= 24
//  fact(5) <= 120
//fact(6) <= 720
720
> > 